website/src/controllers/api-v3/tasks.js

```diff
@@ -54,26 +82,32 @@ api.createTask = {
     toSave = toSave.map(task => task.save({
       validateBeforeSave: false,
     }));
-    toSave.unshift(user.save());
-    let results = await Q.all(toSave);
+    toSave.unshift((challenge || user).save());
+
+    let tasks = await Q.all(toSave);
 
-    if (results.length === 2) { // Just one task created
-      res.respond(201, results[1]);
+    if (tasks.length === 2) {
+      res.respond(201, tasks[1]);
     } else {
-      results.splice(0, 1); // remove the user
-      res.respond(201, results);
+      tasks.splice(0, 1); // remove the user/challenge
+      res.respond(201, tasks);
     }
+
+    // If adding tasks to a challenge -> sync users
+    if (challenge) challenge.addTasks(tasks); // TODO catch/log
   },
 };
 
 /**
- * @api {get} /tasks Get an user's tasks
+ * @api {get} /tasks/:tasksOwner/:challengeId Get an user's tasks
```

Comment:
I assume this is incorrect, given that we're using query parameters here instead?